diff --git a/src/impl/peerconnection.cpp b/src/impl/peerconnection.cpp
--- a/src/impl/peerconnection.cpp
+++ b/src/impl/peerconnection.cpp
@@ -174 +174,7 @@ shared_ptr<IceTransport> PeerConnection::initIceTransport() {
-						    initDtlsTransport();
+						    mProcessor.enqueue([weak_this] {
+							    if (auto locked = weak_this.lock()) {
+								    std::lock_guard signalingLock(locked->signalingMutex);
+								    if (locked->remoteDescription())
+									    locked->initDtlsTransport();
+							    }
+						    });
@@ -1188,0 +1195,6 @@ void PeerConnection::processRemoteDescription(Description description) {
+	if (!dtlsTransport) {
+		auto iceTransport = std::atomic_load(&mIceTransport);
+		if (iceTransport && (iceTransport->state() == Transport::State::Connected ||
+		                     iceTransport->state() == Transport::State::Completed))
+			dtlsTransport = initDtlsTransport();
+	}
